share/ipld/nmt.go

```diff
@@ -65,212 +66,67 @@ func GetNode(ctx context.Context, bGetter blockservice.BlockGetter, root cid.Cid
-			return nil, nil, err
-		}
-		return &ipld.Link{Cid: left}, path[1:], nil
-	case "1":
-		right, err := CidFromNamespacedSha256(n.r)
-		if err != nil {
-			return nil, nil, err
-		}
-		return &ipld.Link{Cid: right}, path[1:], nil
-	default:
-		return nil, nil, errors.New("invalid path for inner node")
-	}
+	panic("method not implemented")
 }
 
 func (n nmtNode) Tree(path string, depth int) []string {
-	if path != "" || depth != -1 {
-		panic("proper tree not yet implemented")
-	}
-
-	return []string{
-		"0",
-		"1",
-	}
+	panic("method not implemented")
 }
 
 func (n nmtNode) ResolveLink(path []string) (*ipld.Link, []string, error) {
-	obj, rest, err := n.Resolve(path)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	lnk, ok := obj.(*ipld.Link)
-	if !ok {
-		return nil, nil, errors.New("was not a link")
-	}
-
-	return lnk, rest, nil
-}
-
-func (n nmtNode) Copy() ipld.Node {
-	l := make([]byte, len(n.l))
-	copy(l, n.l)
-	r := make([]byte, len(n.r))
-	copy(r, n.r)
-
-	return &nmtNode{
-		cid: n.cid,
-		l:   l,
-		r:   r,
-	}
-}
-
-func (n nmtNode) Links() []*ipld.Link {
-	leftCid := MustCidFromNamespacedSha256(n.l)
-	rightCid := MustCidFromNamespacedSha256(n.r)
-
-	return []*ipld.Link{{Cid: leftCid}, {Cid: rightCid}}
+	panic("method not implemented")
 }
 
 func (n nmtNode) Stat() (*ipld.NodeStat, error) {
-	return &ipld.NodeStat{}, nil
+	panic("method not implemented")
 }
 
 func (n nmtNode) Size() (uint64, error) {
-	return 0, nil
-}
-
-type nmtLeafNode struct {
-	cid  cid.Cid
-	Data []byte
-}
-
-func NewNMTLeafNode(id cid.Cid, data []byte) ipld.Node {
-	return &nmtLeafNode{id, data}
-}
-
-func (l nmtLeafNode) RawData() []byte {
-	return append([]byte{nmt.LeafPrefix}, l.Data...)
-}
-
-func (l nmtLeafNode) Cid() cid.Cid {
-	return l.cid
-}
-
-func (l nmtLeafNode) String() string {
-	return fmt.Sprintf(`
-leaf {
-	hash: 		%x,
-	len(Data): 	%v
-}`, l.cid.Hash(), len(l.Data))
-}
-
-func (l nmtLeafNode) Loggable() map[string]interface{} {
-	return nil
-}
-
-func (l nmtLeafNode) Resolve(path []string) (interface{}, []string, error) {
-	return nil, nil, errors.New("invalid path for leaf node")
-}
-
-func (l nmtLeafNode) Tree(_path string, _depth int) []string {
-	return nil
-}
-
-func (l nmtLeafNode) ResolveLink(path []string) (*ipld.Link, []string, error) {
-	obj, rest, err := l.Resolve(path)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	lnk, ok := obj.(*ipld.Link)
-	if !ok {
-		return nil, nil, errors.New("was not a link")
-	}
-	return lnk, rest, nil
-}
-
-func (l nmtLeafNode) Copy() ipld.Node {
-	panic("implement me")
-}
-
-func (l nmtLeafNode) Links() []*ipld.Link {
-	return []*ipld.Link{{Cid: l.Cid()}}
+	panic("method not implemented")
 }
 
-func (l nmtLeafNode) Stat() (*ipld.NodeStat, error) {
-	return &ipld.NodeStat{}, nil
+func (n nmtNode) left() []byte {
+	return n.RawData()[:nmtHashSize]
 }
 
-func (l nmtLeafNode) Size() (uint64, error) {
-	return 0, nil
+func (n nmtNode) right() []byte {
+	return n.RawData()[nmtHashSize:]
```

Comment:
general Q: I don't see this as an issue bc shareSize > nmtHashSize+nmtHashSize but should these methods return nil if nmtNode is leaf? 

I know they're only called on innerNode but wondering if we should add in an additional check.

Reply:
removed those funcs in favour of subslicing directly, they had single call each.